src/azul/http.py

```diff
@@ -386,3 +386,25 @@ def urlopen(self,
                         return response
             else:
                 return response
+
+
+def parse_header(header: str) -> tuple[str, dict[str, str]] | None:
+    """
+    Parse content-type like headers and return a tuple of a key and dictionary
+    pair.
+    """
+    from email.message import (
+        Message,
+    )
+    m = Message()
+    m['content-type'] = header
+    key_and_content: list[tuple[str, str]] | None = m.get_params()
+    if key_and_content is not None:
```

Comment:
I don't believe it is possible for `m.get_params()` to return None if `m['content-type']` has been set, even if it was set to an empty string or None. Maybe this check should be an assert instead?

```
>>> from email.message import Message
... msg = Message()
... msg['content-type'] = None
... print(msg.get_params())
[('', '')]
```

Also, why not just return `(msg.get_content_type(), msg['content-type'].params)` as suggested by the [documentation](https://docs.python.org/3.11/library/cgi.html#cgi.parse_header) about the deprecation of `cgi.parse_header()` ?

Reply:
Didn't get the example from the removed module's doc, but form the [PEP](https://peps.python.org/pep-0594/#cgi).

Opted for a variant of what was in the docs you've pointed out (thanks). The example there wasn't mypy compliant.

Reply:
However, the approach doesn't seem effective for our purposes. https://github.com/DataBiosphere/azul/actions/runs/16635339041/job/47074751299

… I'll sticking to the method I know already works and is discussed in the PEP.